src/perl/Permabit/BlockDevice/VDO.pm

```diff
@@ -1366,6 +1380,20 @@ sub check {
   $self->assertPerformanceExpectations();
 }
 
+########################################################################
+# @inherit
+##
+sub getModuleName {
+  my ($self) = assertNumArgs(1, @_);
+
+  # If useUpstreamModule is set, we need to use $VDO_UPSTREAM_MODNAME(dm-vdo)
+  # module otherwise we load the $VDO_MODNAME(kvdo)
+  if ($self->{useUpstreamModule}) {
+    return $VDO_UPSTREAM_MODNAME;
+  }
+  return $self->SUPER::getModuleName();
+}
+
```

Comment:
This looks right, but I would move this function up the file so that it is next to getModuleVersion() (around line 537).